src/main/io/serial_1wire.c

```diff
@@ -0,0 +1,240 @@
+/*
+ * This file is part of Cleanflight.
+ *
+ * Cleanflight is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Cleanflight is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Ported from https://github.com/4712/BLHeliSuite/blob/master/Interfaces/Arduino1Wire/Source/Arduino1Wire_C/Arduino1Wire.c
+ *  by Nathan Tsoi <nathan@vertile.com>
+ */
+
+#include <stdbool.h>
+
+#include "platform.h"
+
+#ifdef USE_SERIAL_1WIRE
+
+#include "drivers/gpio.h"
+#include "drivers/inverter.h"
+#include "drivers/light_led.h"
+#include "drivers/system.h"
+#include "io/serial_1wire.h"
+
+const escHardware_t escHardware[ESC_COUNT] = {
+// Figure out esc clocks and pins, extrapolated from timer.c
+// Periphs could be pulled progmatically... but I'll leave that for another exercise
+#if defined(STM32F3DISCOVERY) && !(defined(CHEBUZZF3))
+  { GPIOD, 12 },
+  { GPIOD, 13 },
+  { GPIOD, 14 },
+  { GPIOD, 15 },
+  { GPIOA, 1 },
+  { GPIOA, 2 }
+#elif defined(CJMCU) || defined(EUSTM32F103RC) || defined(NAZE) || defined(OLIMEXINO) || defined(PORT103R)
```

Comment:
CJMCU is an FC with brushed motor outputs. The definition here is useless. The ports for the motor outputs are also quite different to the Naze. The Naze can have up to 8 motors (esc's) connected.